Fix toolbar keys order in RTL layout

codokie opened the pull request:
Fix the toolbar keys direction so that it does not change when switching between an RTL/LTR layouts.
The issue was described here: https://github.com/Helium314/HeliBoard/issues/557
Upon testing the app in Android Studio, the order seems to be fixed with no adverse effect on the direction of the text in the auto-complete (RTL layout) suggestions.

codokie commented:
Commit [1eac01e](https://github.com/Helium314/HeliBoard/pull/574/commits/1eac01e8d71ad066b33f51a30f510a2c4c77eb7c) has a reference to PR number 2, but it is unrelated to it.
Please excuse me as I am new to GitHub, apparently commit messages may not be edited.

Helium314 commented:
> Please excuse me as I am new to GitHub, apparently commit messages may not be edited.

That's normal in git, and usually they should not be edited.
Having additional "useless" commits is not an issue, when merging I squash everything into a single commit anyway.

I see you didn't remove `setRtl`, it still adjusts the layout direction of `mStripVisibilityGroup`. Is this useful / needed, or can it be removed too?

Did you test the arrow keys in the toolbar, and when pinned? Just to be sure they point in the correct direction.

> in case quirks that exist in GBoard are to be kept in HeliBoar, I would be grateful if someone could implement a UI toggle in settings to make the toolbar direction fixed for RTL languages.

I'm ok with merging this without toggle, but would you be willing to try adding one later, in case someone complains about this change?

Helium314 commented on app/src/main/java/helium314/keyboard/latin/suggestions/SuggestionStripView.java:
```diff
@@ -129,7 +128,7 @@ public StripVisibilityGroup(final View suggestionStripView,
             final int layoutDirection = isRtlLanguage ? ViewCompat.LAYOUT_DIRECTION_RTL
                     : ViewCompat.LAYOUT_DIRECTION_LTR;
-            ViewCompat.setLayoutDirection(mSuggestionStripView, layoutDirection);
+            ViewCompat.setLayoutDirection(mSuggestionStripView, ViewCompat.LAYOUT_DIRECTION_LOCALE);
             ViewCompat.setLayoutDirection(mSuggestionsStrip, layoutDirection);
```
You set `mSuggestionStripView` to follow layout direction of the system locale, but `mSuggestionsStrip` to follow the keyboard locale.
Without testing myself, I'm not sure what this actually does,. Can you elaborate on why?

codokie replied:
From my testing, 'mSuggestionStripView' is responsible for the toolbar overall direction. Changing the layout direction of 'mSuggestionsStrip' seemed to only affect the direction of the vertical bars '|' that separate each suggested word.
Nevertheless I think it may be a good idea to make them both follow the same direction like they previously did, so I changed that in the new commit.

codokie commented:
> Having additional "useless" commits is not an issue, when merging I squash everything into a single commit anyway.

That's neat, I did not know you could combine those together. Thanks for the reassurance :)
> would you be willing to try adding one later, in case someone complains about this change?

So I actually learned how to add a toggle myself, it was not really complicated. I added the toggle to the Preferences screen.
(I did some testing and so far it seems to work fine..)
>Did you test the arrow keys in the toolbar, and when pinned?

Arrow keys/pinned toolbar buttons point in the same direction as before if the toggle is off.
If it is turned on, then the direction of all toolbar items is determined by the system locale.
>I see you didn't remove setRtl, it still adjusts the layout direction of mStripVisibilityGroup. Is this useful / needed, or can it be removed too?

In the [845e6f9](https://github.com/Helium314/HeliBoard/pull/574/commits/845e6f976763eeed3804433605e7b0ef18beef98) commit I've moved some of the code of the setLayoutDirection method to setRtl, because the former belongs to a static class and the mRtl variable (which has been restored) could not be referenced from it. So setRtl is still needed I think.

Helium314 commented:
Thanks, this is fine now.

My only issue is with the name and description of the setting. The name is ambiguous, and it's completely unclear that the setting is about ltr / rtl languages.

Helium314 commented:
Maybe something like "toolbar direction language direction" (though I'm not really happy with that one as well).

jermanuts commented:
what about "toolbar direction"?

Helium314 commented:
I would like to avoid RTL / LTR shortcuts as it would be confusing to many people.

How about reversing the setting and use _Variable toolbar direction_, with some description like _Reverse toolbar_ (or toolbar direction?) _for right-to-left scripts_ (languages?)
